### PR TITLE
Make index sets configurable via Graylog REST API

### PR DESCRIPTION
**Pull request by joschi**
## Description

Make index sets configurable via Graylog REST API

Refs #2880

## Types of changes

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog2/indexer/indexset/IndexSetConfig.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+    @JsonProperty("description")
+    @Nullable
+    public abstract String description();
+
+    @JsonProperty("index_prefix")
+    @NotBlank
+    public abstract String indexPrefix();
+
+    @JsonProperty("shards")
```
Comment (bernd) [inline graylog2-server/src/main/java/org/graylog2/indexer/indexset/IndexSetConfig.java]: Yesss! :+1:

File: graylog2-server/src/main/java/org/graylog2/rest/resources/system/indexer/IndexSetsResource.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    @GET
+    @Timed
+    @ApiOperation(value = "Get a list of all index sets")
+    @ApiResponses(value = {
+            @ApiResponse(code = 403, message = "Unauthorized"),
+    })
+    public IndexSetResponse list() {
+        final Set<IndexSetConfig> all = indexSetService.findAll();
+        final Set<IndexSetSummary> indexSets = new HashSet<>(all.size());
+        for (IndexSetConfig indexSetConfig : all) {
```
Comment (bernd) [inline graylog2-server/src/main/java/org/graylog2/rest/resources/system/indexer/IndexSetsResource.java]: Out of interest: Any reason you are using functional idioms in `get()` but not here? :smiley:

File: graylog2-server/src/main/java/org/graylog2/rest/resources/system/indexer/responses/IndexSetResponse.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.auto.value.AutoValue;
+
+import java.util.Set;
+
+@JsonAutoDetect
+@AutoValue
+public abstract class IndexSetResponse {
+    @JsonProperty("total")
```
Comment (bernd) [inline graylog2-server/src/main/java/org/graylog2/rest/resources/system/indexer/responses/IndexSetResponse.java]: :+1: for using a wrapper object here. Makes implementing pagination easy in the future.

File: graylog2-server/src/main/java/org/graylog2/rest/resources/system/indexer/IndexSetsResource.java (hunk 2 above)
Comment (joschi) [inline graylog2-server/src/main/java/org/graylog2/rest/resources/system/indexer/IndexSetsResource.java]: Heh, no particular reason. It seems I was just in "zombie" mode. 😉 

I'll change it for consistency.

**Pull request merged**